core/modules/ccontrol/testAntlr4GeneratedIR.cc

```diff
@@ -0,0 +1,342 @@
+query::OrderByTerm OrderByTerm(shared_ptr<query::ValueExpr> const& term) {
+    return query::OrderByTerm(term);
+}
+
+
+/// Create a new OrTerm. Args can be a shared_ptr to any kind of object that inherits from BoolTerm.
+template <typename...Targs>
+shared_ptr<query::OrTerm> OrTerm(Targs... args) {
+    vector<shared_ptr<query::BoolTerm>> terms;
+    pusher<shared_ptr<query::BoolTerm>, shared_ptr<query::BoolTerm>>(terms, args...);
+    return make_shared<query::OrTerm>(terms);
+}
+
+
+/// Create a new PassTerm with given text.
+shared_ptr<query::PassTerm> PassTerm(string const& text) {
+    return make_shared<query::PassTerm>(text);
+}
+
+
+/// Create a new SelectList. Args should be a comma separated list of shared_ptr to ValueExpr.
+template <typename...Targs>
+shared_ptr<query::SelectList> SelectList(Targs... args) {
+    auto ptr = make_shared<vector<shared_ptr<query::ValueExpr>>>();
+    pusher(*ptr, args...);
+    return make_shared<query::SelectList>(ptr);
+}
+
+
+/// Create a new SelectList with the given members.
+shared_ptr<query::SelectStmt> SelectStmt(shared_ptr<query::SelectList> const& selectList,
+        shared_ptr<query::FromList> const& fromList,
+        shared_ptr<query::WhereClause> const& whereClause,
+        shared_ptr<query::OrderByClause> const& orderByClause) {
+    return make_shared<query::SelectStmt>(selectList, fromList, whereClause, orderByClause);
+}
+
+
+/// Create a new TableRef with the given database, table, and alias names.
+shared_ptr<query::TableRef> TableRef(string const& db, string const& table,
+        const string& alias) {
+    return make_shared<query::TableRef>(db, table, alias);
+}
+
+
+/// Create a new ValueExpr with a ValueFactorPtr.
+shared_ptr<query::ValueExpr> ValueExpr(shared_ptr<query::ValueFactor> const& valueFactor) {
+    // todo, allow pusher to take a func that returns a T, where we can pass the valueFactor
+    // to a new FactorOp ptr, and push that ptr into the vector.
+    vector<query::ValueExpr::FactorOp> factorOps = {query::ValueExpr::FactorOp(valueFactor)};
+    return make_shared<query::ValueExpr>(factorOps);
+}
+
+
+/// Create a ValueFactor with a COLUMNREF value.
+shared_ptr<query::ValueFactor> ValueFactor(shared_ptr<query::ColumnRef> const& columnRef) {
+    return make_shared<query::ValueFactor>(columnRef);
+}
+
+
+/// Create a ValueFactor with a CONST value.
+shared_ptr<query::ValueFactor> ValueFactor(string const& constVal) {
+    return make_shared<query::ValueFactor>(constVal);
+}
+
+
+/// Create a new WhereClause with a given OrTerm for its root term.
+shared_ptr<query::WhereClause> WhereClause(shared_ptr<query::OrTerm> const& orTerm) {
+    return make_shared<query::WhereClause>(orTerm);
+}
+
+
+struct Antlr4TestQueries {
+    Antlr4TestQueries(string const& iQuery, shared_ptr<query::SelectStmt> const& iCompareStmt,
+            string const& iSerializedQuery)
+    : query(iQuery)
```

Comment:
Member initialization is better to be indented by 4 spaces (even though there i uno such explicit requirement in the Style Guide). This should increase the readability of the code:
```
    Antlr4TestQueries(string const& iQuery, shared_ptr<query::SelectStmt> const& iCompareStmt,
            string const& iSerializedQuery)
        : query(iQuery)
        , compareStmt(iCompareStmt)
        , serializedQuery(iSerializedQuery)
    {}
```